name collision with macro DOMAIN
 $ ggrep -rw DOMAIN /usr/include/
/usr/include/math.h:#define     DOMAIN          1

Need to be evaluated before submitting to upstream.

################################################################################
/builds/jsasek/squid/components/squid/squid-4.10/src/esi/VarState.cc: In member function 'char* ESIVariableUserAgent::getProductVersion(cons
t char*)':
/builds/jsasek/squid/components/squid/squid-4.10/src/esi/VarState.cc:153:9: error: 'index' was not declared in this scope
     t = index(s,'/');
         ^~~~~
/builds/jsasek/squid/components/squid/squid-4.10/src/esi/VarState.cc:153:9: note: suggested alternative: 'index_t'
     t = index(s,'/');
         ^~~~~
         index_t

--- squid-4.8/src/auth/basic/SMB_LM/valid.h	2019-07-31 10:19:39.629726916 +0000
+++ squid-4.8/src/auth/basic/SMB_LM/valid.h	2019-07-31 10:17:54.496968865 +0000
@@ -15,7 +15,7 @@
 #define NTV_PROTOCOL_ERROR 2
 #define NTV_LOGON_ERROR 3
 
-int Valid_User(char *USERNAME, char *PASSWORD, const char *SERVER, char *BACKUP, const char *DOMAIN);
+int Valid_User(char *USERNAME, char *PASSWORD, const char *SERVER, char *BACKUP, const char *Domain);
 
 #endif
 
--- squid-4.8/src/auth/basic/SMB_LM/valid.cc	2019-07-31 10:19:43.444978870 +0000
+++ squid-4.8/src/auth/basic/SMB_LM/valid.cc	2019-07-31 10:18:54.616077251 +0000
@@ -22,7 +22,7 @@
 
 // BACKUP is unused
 int
-Valid_User(char *USERNAME, char *PASSWORD, const char *SERVER, char *, const char *DOMAIN)
+Valid_User(char *USERNAME, char *PASSWORD, const char *SERVER, char *, const char *Domain)
 {
     const char *supportedDialects[] = {"PC NETWORK PROGRAM 1.0",
                                        "MICROSOFT NETWORKS 1.03",
@@ -37,7 +37,7 @@ Valid_User(char *USERNAME, char *PASSWOR
     SMB_Handle_Type con;
 
     SMB_Init();
-    con = SMB_Connect_Server(NULL, SERVER, DOMAIN);
+    con = SMB_Connect_Server(NULL, SERVER, Domain);
     if (con == NULL) {
         return (NTV_SERVER_ERROR);
     }
--- squid-4.8/src/esi/VarState.cc	2019-07-31 10:46:04.726002599 +0000
+++ squid-4.8/src/esi/VarState.cc	2019-07-31 10:37:51.121231505 +0000
@@ -12,6 +12,7 @@
 #include "esi/VarState.h"
 #include "fatal.h"
 #include "HttpReply.h"
+#include <strings.h>
 
 char const *ESIVariableUserAgent::esiUserOs[]= {
     "WIN",
